### PR TITLE
feat: add an hours formatter

### PR DESCRIPTION
**Pull request by pepicrft**
## Summary
- Extract `formatHours` function from Chart.js into a dedicated `formatters.js` module
- Add comprehensive Vitest test suite with 14 test cases covering all scenarios
- Update Chart.js to import `formatHours` from the new module
- Add npm test script and mise task for running tests

## Test plan
- [x] All existing functionality preserved - formatHours works identically
- [x] Comprehensive test coverage with 14 test cases covering:
  - Hours less than 1 (formatted as minutes)
  - Hours between 1-24 (formatted as hours and minutes) 
  - Hours 24+ (formatted as days and hours)
  - Edge cases and rounding scenarios
- [x] Build process works correctly with new import structure
- [x] All tests pass: `npm test`
- [x] Build succeeds: `npm run build`

### CODE REVIEW

File: web/js/Chart.js [hunk 1]
```diff
@@ -38,34 +39,6 @@ function formatMilliseconds(milliseconds) {
   return formatSeconds(milliseconds / 1000);
 }
 
-/**
- * Formats hours into a human readable string
- * @param {number} hours - The time duration in hours
- * @returns {string} Formatted time string (e.g., "30m", "2h 30m", "1d 5h")
- */
-function formatHours(hours) {
```
Comment (pepicrft) [inline web/js/Chart.js:46]: I committed this by mistake to `main` 🤦🏼. I refrained from reverting, and instead created this PR adding tests.

**Review by cschmatzler — APPROVED** [review]
